src/physical/tabular/operations/triescan_select.rs

```diff
@@ -126,12 +128,10 @@ impl<'a> TrieScanSelectEqual<'a> {
 }
 
 impl<'a> TrieScan<'a> for TrieScanSelectEqual<'a> {
-    #[allow(clippy::unnecessary_lazy_evaluations)] // not actually
-                                                   // unnecessary, as the subtraction might underflow
     fn up(&mut self) {
         debug_assert!(self.current_layer.is_some());
 
-        self.current_layer = self.current_layer.and_then(|index| index.checked_sub(1));
+        self.current_layer = self.current_layer.unwrap().checked_sub(1);
```

Comment:
What about `unwrap_or_default()` here, which would potentially avoid a panic?

Reply:
As per assertion above it should never panic.

Reply:
Then we can get rid of the assert and just use `expect(…)`, which will at least give us a nice error message, since we're doing the check anyways.